public/js/TableBuilder.js

```diff
@@ -0,0 +1,63 @@
+class TableBuilder {
+  constructor(body) {
+    this.body = body;
+  }
+
+  createRow() {
+    let row = document.createElement('tr');
+    this.body.appendChild(row);
+    return row;
+  }
+
+  createTextOnlyCell(row, textContent) {
+    let cell = document.createElement('td');
+    cell.textContent = textContent;
+    row.appendChild(cell);
+    return cell;
+  }
+
+  createButtonCell(row, buttonText, id, buttonFunction) {
+    let cell = document.createElement('td');
+    row.appendChild(cell);
+
+    let button = document.createElement('button');
+    button.setAttribute("id", buttonText + id);
```

Comment:
I don't like that we are making assumptions about how the client may want to set the ID of the newly created button. This function returns the cell object created. The client can do something like this to set the ID of the button.
```
let tb = new TableBuilder(recipe_table_body);
let row = tb.createRow();
let new_cell = tb.createButtonCell(row, "Remove", removeFunction);
let new_button = new_cell.children[0];
new_button.setAttribute("id", "Remove" + id);
```